### PR TITLE
added academic year dropdown filter to search

### PR DESCRIPTION
**Pull request by djvoa12**
**Summary:**
Added academic year dropdown filter for search.

**UI:**
cancer

### CODE REVIEW

File: app/components/global-search.js [hunk 1]
```diff
@@ -32,23 +34,25 @@ export default Component.extend({
     return new Headers(headers);
   }),
 
-  paginatedResults: computed('page', 'results', 'size', function() {
+  filteredResults: computed('results', 'selectedYear', function() {
```
Comment (jrjohnson) [inline app/components/global-search.js]: Should be `results.[]` I think.

File: app/components/global-search.js [hunk 2]
```diff
@@ -32,23 +34,25 @@ export default Component.extend({
...
+    return this.results ? this.results.filterBy('year', this.selectedYear) : [];
+  }),
+
+  paginatedResults: computed('filteredResults', 'page', 'size', function() {
```
Comment (jrjohnson) [inline app/components/global-search.js]: should be `filteredResults.[]`

File: app/components/global-search.js [hunk 3]
```diff
@@ -59,4 +63,17 @@ export default Component.extend({
...
   },
+
+  actions: {
+    setSelectedYear(year) {
+      this.set('selectedYear', parseInt(year, 10));
+      this.onSelectPage(1);
+    }
+  },
+
+  setUpYearFilter(years) {
```
Comment (jrjohnson) [inline app/components/global-search.js]: This should be a computed property based on search results.

File: tests/pages/components/global-search.js [hunk 4]
```diff
@@ -1,15 +1,22 @@
...
+  value
 } from 'ember-cli-page-object';
 
 const definition = {
   scope: '[data-test-global-search]',
   noResultsIsVisible: isVisible('.no-results'),
   input: fillable('input.global-search-input'),
-  clickIcon: clickable('[data-test-search-icon]')
+  clickIcon: clickable('[data-test-search-icon]'),
+  academicYear: value('select'),
```
Comment (jrjohnson) [inline tests/pages/components/global-search.js]: this is no specific enough. Please put a `[data-test-academic-year-filter]` attribute on the select element so it's clear what the test is using.

File: app/controllers/search.js [hunk 5]
```diff
@@ -7,5 +7,11 @@ export default Controller.extend({
   },
 
   page: 1,
-  query: ''
+  query: '',
+
+  actions: {
+    setQuery(query) {
+      this.setProperties({ page: 1, query });
```
Comment (jrjohnson) [inline app/controllers/search.js]: Unrelated bugfix. Should go in it's own PR.

Comment (djvoa12) [reply app/controllers/search.js]: Moved.

File: tests/pages/components/global-search.js (hunk 4 above)
Comment (djvoa12) [reply tests/pages/components/global-search.js]: Done.

File: app/templates/components/global-search.hbs [hunk 6]
```diff
@@ -1,6 +1,24 @@
...
+      <label>{{t "general.academicYear"}}:</label>
+      <select
+        data-test-academic-year-filter
+        onchange={{action "setSelectedYear" value="target.value"}}>
+        <option selected={{is-equal null this.selectedYear}} value="">
+          {{t "general.allAcademicYears"}}
+        </option>
+        {{#each this.yearOptions as |year|}}
+          <option selected={{is-equal year this.selectedYear}} value={{year}}>
+            {{year}} - {{inc year}}
```
Comment (jrjohnson) [inline app/templates/components/global-search.hbs]: NICE!
